lib/compilers/dosbox-compiler.ts

```diff
@@ -0,0 +1,175 @@
+
+        if (files && files.length > 0) {
+            (filters as any).dontMaskFilenames = true;
+
+            await this.writeMultipleFiles(files, dirPath);
+        }
+
+        return {
+            inputFilename,
+        };
+    }
+
+    private getDosboxArgs(tempDir: string, compileArgs: string[]) {
+        const binPath = path.relative(this.root, path.dirname(this.compiler.exe));
+        const exeName = path.basename(this.compiler.exe).replace(/\.exe$/i, '');
+        return [
+            '-c',
+            `mount c ${this.root}`,
+            '-c',
+            `mount d ${tempDir}`,
+            '-c',
+            `PATH=%PATH%;C:\\${binPath}`,
+            '-c',
+            'd:',
+            '-c',
+            `${exeName} ${compileArgs.join(' ')} > STDOUT.TXT`,
+            '-c',
+            'exit',
+        ];
+    }
+
+    private getDosboxEnv() {
+        return {
+            SDL_VIDEODRIVER: 'dummy',
+        };
+    }
+
+    protected override async execCompilerCached(compiler, args, options) {
+        if (this.mtime === null) {
+            throw new Error('Attempt to access cached compiler before initialise() called');
+        }
+        if (!options) {
+            options = this.getDefaultExecOptions();
+            options.timeoutMs = 0;
+            options.ldPath = this.getSharedLibraryPathsAsLdLibraryPaths([]);
+        }
+
+        const key = this.getCompilerCacheKey(compiler, args, options);
+        let result = await this.env.compilerCacheGet(key);
+        if (!result) {
+            result = await this.env.enqueue(async () => this.exec(compiler, args, options));
+            if (result.okToCache) {
+                this.env
+                    .compilerCachePut(key, result)
+                    .then(() => {
+                        // Do nothing, but we don't await here.
+                    })
+                    .catch(e => {
+                        logger.info('Uncaught exception caching compilation results', e);
+                    });
+            }
+        }
+
+        return result;
+    }
+
+    public override async exec(filepath: string, args: string[], execOptions: any) {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        execOptions.env = this.getDosboxEnv();
+
+        if (!execOptions.customCwd) {
+            execOptions.customCwd = await this.newTempDir();
+        }
+
+        const tempDir = execOptions.customCwd;
+        const fullArgs = this.getDosboxArgs(tempDir, args);
+
+        const result = await exec.executeDirect(this.dosbox, fullArgs, execOptions);
+
+        const stdoutFilename = path.join(tempDir, 'STDOUT.TXT');
+        const stdout = await fs.readFile(stdoutFilename);
+        (result as any).stdout = stdout.toString('utf8');
+
+        return result;
+    }
+
+    public override async runCompiler(compiler, options, inputFilename, execOptions) {
+        if (!execOptions) {
```

Comment:
I _think_ this can be implemented in terms of the base class:

```
return super.runCompiler(..., options.map(...), ...);
```

I think? (like, just pass the superclass (syntax may be broken) the modified options but otherwise let it do its thing

Reply:
done, seemed to work